RocketChat/Rocket.Chat
# fix: Public channels in private teams are visible in spotlight

## Proposed changes (including videos or screenshots)


This pull request fixes the visibility of public channels belonging to private teams when user is searching.

Image of search for member of private team :-
![Screenshot 2023-04-29 024051](https://user-images.githubusercontent.com/89914602/235257513-07887ec0-12ea-4cd0-9e6b-f8987f761a04.png)

Image of search for non-members of private team :-
![Screenshot 2023-04-29 024119](https://user-images.githubusercontent.com/89914602/235257730-034e8de2-71c2-471e-a52a-3b26de410ddc.png)



## Issue(s)

fix #28982 
## Steps to test or reproduce

1. Create a private Team
2. Create a public channel inside the Team
3. Login as an user that is not a member of that Team
4. Open the search (^+K) and search for that channel
5. Observe that channel is not visible
